### DISCUSSION TITLE
Clarify usage of `.form-label` with `.form-floating`

### DISCUSSION THREAD

**Discussion by imagoiq** [twbs/bootstrap#39971] [Q&A] [answered]
Hi,

According to the [Migration guide of v5.0,](https://getbootstrap.com/docs/5.3/migration/) "Form labels now require .form-label.". But when looking at the documentation page of [Floating labels](https://getbootstrap.com/docs/5.3/forms/floating-labels/), no label use this class.

I also encountered a bug with [Readonly plaintext](https://getbootstrap.com/docs/5.3/forms/floating-labels/#readonly-plaintext) where if you change the `$input-bg` variable to something different from the body background, the label background is visible.

```html
<div class="form-floating">
    <input type="text" class="form-control-plaintext" readonly value="my value" id="a-field">
    <label class="form-label" for="a-field">My label</label>
</div>
```

![image](https://github.com/twbs/bootstrap/assets/12294151/2c14b996-38fd-40e4-8c26-7846bdad899e)

So now I'm totally confused. Should I report a bug for the issue above? Is the doc page missing the `.form-label`? Is `.form-floating` label compatible with `.form-label`?

**Comment by julien-deramond** [comment, marked as answer]
Hello @imagoiq 

> "Form labels now require .form-label.". 
> ...
> So now I'm totally confused. Should I report a bug for the issue above? Is the doc page missing the .form-label? Is .form-floating label compatible with .form-label?

`.form-label` are used for regular form controls (e.g. https://getbootstrap.com/docs/5.3/forms/form-control/). Floating labels are particular and don't require any extra `.form-label` class since it's handled directly by `.form-floating` put on the surrounding `<div>`.

> I also encountered a bug with [Readonly plaintext](https://getbootstrap.com/docs/5.3/forms/floating-labels/#readonly-plaintext) where if you change the $input-bg variable to something different from the body background, the label background is visible.

I'm not entirely sure whether this is an issue, it seems rather consistent and acceptable to have the label background visible here.

**Reply by imagoiq** [reply]
Thanks for the clarification. It would be nice to precise it in the migration doc page, I think. I worked with several different teams and I saw this mix of classes multiple time.

**Reply by julien-deramond** [reply]
Thank you for the feedback. I'll add that to my to-do list and try to find a way to make it more understandable.
